roles/ceph-osd/tasks/activate_osds.yml

```diff
@@ -69,7 +69,7 @@
     is_before_infernalis
 
 - name: get osd id (for or after infernalis)
-  shell: "ls /var/lib/ceph/osd/ |grep -oh '[0-9]*'"
+  shell: "ls /var/lib/ceph/osd/ | grep '-' | cut -d '-' -f 2"
```

Comment:
The first `grep` is not needed as cut already looks for the second field :).
So you can just keep the cut.